src/System.ObjectModel/tests/ObservableCollection/ObservableCollection_MethodsTest.cs

```diff
@@ -454,6 +454,180 @@ public static void GetEnumeratorTest()
             Assert.Equal(col.Count, i);
             e.Dispose();
         }
+
+        [Fact]
+        public static void InsertRange_NotifyCollectionChanged_Beginning_Test()
+        {
+            int[] dataToInsert = new int[] { 1, 2, 3, 4, 5 };
+            int[] initialData = new int[] { 10, 11, 12, 13 };
+            int eventCounter = 0;
+            ObservableCollection<int> collection = new ObservableCollection<int>(initialData);
+            collection.CollectionChanged += (o, e) => eventCounter++;
+
+            collection.InsertRange(0, dataToInsert);
+
+            Assert.NotNull(collection);
+            Assert.Equal(dataToInsert.Length + initialData.Length, collection.Count);
+            Assert.Equal(dataToInsert.Length, eventCounter);
+            Assert.Equal(dataToInsert[0], collection[0]);
```

Comment:
nit:
```C#
Assert.True(dataToInsert.AsSpan().SequenceEqual(collection.AsSpan(0, 4));
Assert.True(initialData.AsSpan().SequenceEqual(collection.AsSpan(5));
```

Reply:
Here/elsewhere.

Alternatively:
```C#
Assert.Equal(dataToInsert, collection.AsSpan(0, 4).ToArray());
Assert.Equal(initialData, collection.AsSpan(5).ToArray());
```

Reply:
@ahsonkhan this is GREAT advice, didn't know I could do that with xUnit and `AsSpan`. Since `Collection<T>` can't use `AsSpan()` I am going to implement this solution:

```c#
int[] collectionAssertion = collection.ToArray();
Assert.Equal(dataToInsert, collectionAssertion.AsSpan(0, 5).ToArray());
Assert.Equal(initialData, collectionAssertion.AsSpan(5).ToArray());
```